components/ui/AppMegaDropdownMenu.vue

```diff
@@ -1,38 +1,63 @@
 <template>
   <article class="app-mega-dropdown">
-    <button
-      ref="button"
-      class="app-mega-dropdown__button"
-      :class="`app-mega-dropdown__button_${kind}`"
-      @click="toggleOpen"
+    <label
+      ref="searchFieldWrapper"
+      class="app-mega-dropdown__search-field-wrapper"
+      :class="`app-mega-dropdown__search-field-wrapper_${kind}`"
     >
-      <span>{{ placeholder }}</span>
-      <svg class="app-mega-dropdown__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M16 22L6 12l1.4-1.4 8.6 8.6 8.6-8.6L26 12z" /></svg>
-    </button>
+      <input
+        v-model="textOnTheFilter"
+        type="text"
+        class="app-mega-dropdown__search-field-wrapper__input"
+        :placeholder="`${placeholder}`"
```

Comment:
This can be simplified:

```suggestion
        :placeholder="placeholder"
```